keras_nlp/tests/test_case.py

```diff
@@ -223,3 +262,155 @@ def run_class_serialization_test(self, instance):
                 if "__annotations__" in lst:
                     lst.remove("__annotations__")
             self.assertEqual(ref_dir, new_dir)
+
+    def run_model_saving_test(
+        self,
+        cls,
+        init_kwargs,
+        input_data,
+    ):
+        """Save and load a model from disk and assert output is unchanged."""
+        model = cls(**init_kwargs)
+        model_output = model(input_data)
+        path = os.path.join(self.get_temp_dir(), "model.keras")
+        model.save(path, save_format="keras_v3")
+        restored_model = keras.models.load_model(path)
+
+        # Check we got the real object back.
+        self.assertIsInstance(restored_model, cls)
+
+        # Check that output matches.
+        restored_output = restored_model(input_data)
+        self.assertAllClose(model_output, restored_output)
+
+    def run_backbone_test(
+        self,
+        cls,
+        init_kwargs,
+        input_data,
```

Comment:
For backbones, do we expect the `input_data` to be the same for all tests? Or is there at least a sensible default?

Reply:
There's not a great default that would really apply all over. There is some variation on what feature inputs these models expect after preprocessing.